### PR TITLE
Documentation: Add contributions page

### PR DESCRIPTION
**Pull request by valighita**
Signed-off-by: Valentin Ghita valx92@gmail.com

### CODE REVIEW

File: Documentation/teaching/index.rst [hunk 1]
```diff
@@ -28,7 +28,7 @@ To get started build the documentation from the sources after
 installing docker-compose on you host:
 
 .. code-block:: c
-   
+
    cd tools/teaching && make docker-docs
```
Comment (tavip) [inline Documentation/teaching/index.rst]: Since you are touching this, can you please change tools/teaching to tools/labs?

Comment (valx90) [reply Documentation/teaching/index.rst]: Fixed.

File: Documentation/teaching/info/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    student@eg106:~/so2/linux$ git commit -m "<commit message>"
+
+  .. warning::
+
+    The commit message must include a relevant description of your change
+    and the location of the changed component.
+
+    Examples:
+
+      * ``documentation: index: Fix typo in the first section``
```
[the remark this answers is not in the document]
Comment (valighita) [reply Documentation/teaching/info/contributing.rst]: I don't know which one is better. @tavip  @dbaluta  ?

[the remark this answers is not in the document]
Comment (dbaluta) [reply Documentation/teaching/info/contributing.rst]: @valighita I've seen both but I think the preferred way is with blanks.

File: Documentation/teaching/info/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+    student@eg106:~/so2/linux$ git commit --amend
+
+4. Force-push the updated commit:
+
+  .. code-block:: bash
+
+    student@eg106:~/so2/linux$ git push my_fork <your_branch_name> -f
+
+  After this step, the Pull Request is updated.
+
```
[the remark this answers is not in the document]
Comment (valighita) [reply Documentation/teaching/info/contributing.rst]: Done.

**Pull request merged**